Sources/MessagingPush/PushHandling/iOSPushEventListener.swift

```diff
@@ -34,7 +28,7 @@ protocol PushEventListener: AutoMockable {
  - During the native iOS SDK's initialization, the SDK's digraph instance is re-created. All objects (and singletons) in that old digraph instance are deleted from memory.
  - That's bad! If the PushEventListener singleton instance was stored in the digraph, it would be deleted from memory. The OS would no longer be able to send push notification callbacks to the SDK.
  */
-class iOSPushEventListener: NSObject, PushEventListener, UNUserNotificationCenterDelegate {
+class iOSPushEventListener: BaseSdkPushEventListener {
```

Comment:
This class now no longer has knowledge of the `UserNotifications` framework. This class contains all logic to handle push click events for our SDK but is also fully testable! All by inheriting to our protocol, `PushEventListener`.